mona_sdk/async_client.py

```diff
@@ -0,0 +1,47 @@
+import asyncio
+from functools import wraps, partial
+from mona_sdk import Client
+
+
+def async_wrap(func):
+    @wraps(func)
+    async def run_inner(*args, **kwargs):
+        async_client = args[0]
+        final_event_loop = (
+            kwargs.pop("event_loop", None)
+            or async_client._event_loop
```

Comment:
should we double check if the async_client is actually of the AsyncClient type and return a better error if that's the case?

Reply:
@nemo-mona I don't think so.. it called only from within the async client itself.